### ISSUE TITLE
Connecting LPX to other devices (Yamaha SEQTRAK)

### ISSUE THREAD

**Issue by leonidaslundell** [mat1jaczyyy/LP-Firmware-Utility#53]
Hi, thanks for the amazing tool! 

I realize this is a big question, but I am not familiar with editing HEX code so I would appreciate any direction!

I recently bought a Yamaha SEQTRAK and was disappointed to realize that the devices do not talk to each directly. I used the midimonitor.com and read the [programmer documentation](https://fael-downloads-prod.focusrite.com/customer/prod/s3fs-public/downloads/Launchpad%20X%20-%20Programmers%20Reference%20Manual.pdf), and it seems that the main difference is that the LPX sends communicates through Note **On** velocity 127, and Note **On** velocity 0, while the SEQTRAK seems to communicate with Note **On** velocity 100, and Note **Off** velocity 0.

This is of course not confirmed, as I am unable to edit the .syx file, or find an easy way to send signals over USB. 

To summarize, is it possible to change the note on/note off behaviour, and if yes, can you point me to workflow?

Thanks!

**Comment by mat1jaczyyy** [comment]
Hey,

how are you attempting to interface with the Yamaha SEQTRAK currently? I'm not familiar with it, but it looks to be a standalone device of some sort. The Launchpad X only offers a USB port and no direct MIDI ports. Is your LP X talking to the computer first, and then the computer talks to the SEQTRAK?

**Comment by leonidaslundell** [comment]
Appreciate the quick reply! 

It's a usb C to usb C connection. There are YT videos of people connecting other novation products to the seqtrak, i think the issue is specific to the communication of the LPX.

**Comment by mat1jaczyyy** [comment]
Ah, so the SEQTRAK is acting as a USB host in this exchange?

Yeah, you'd need to hack on the firmware to change these send values. I don't know the exact points in the firmware, you'll need to do the reverse engineering yourself, you can use our ghidra stuff as a starting point I guess...

The firmware binary to edit is [here](https://github.com/mat1jaczyyy/LP-Firmware-Utility/blob/master/firmware/stock/LPX-351.bin) and you can use [bintosyx](https://github.com/mat1jaczyyy/LP-Firmware-Utility/tree/master/tools/bintosyx) (build with `make tools`) to convert it into an acceptable firmware update.

@gabonator has done significantly more sophisticated hacks on the Mini MK3's firmware (very similar to X firmware), so you might have some luck reading through the source code [here](https://github.com/gabonator/Projects/tree/master/NovationLaunchpad).

**Comment by leonidaslundell** [comment]
Super, thanks for pointing me in the right direction!

 What do you think @gabonator? Is this doable or a nightmare fools errand?
